oauth2_client/providers/labster/provider.py

```diff
@@ -0,0 +1,50 @@
+"""
+LabsterProvider
+Currently supports only Client Credentials OAuth2 flow.
+"""
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+from oauth2_client.models import AccessToken
+from oauth2_client.providers.oauth2.provider import OAuth2Provider
+
+
+class LabsterProvider(OAuth2Provider):
+    """
+    Labster OAuth2 provider
+    """
+    id = 'labster'
+    name = 'Labster'
+
+    def accesstoken_from_raw_token(self, raw_token):
+        """
+        The raw token returned by methods from the `fetch_token_*` familty is provider-specific.
+        This method converts the raw token to a unified data model used by the oauth2_client.
+
+        :param raw_token: dict
+        :return: oauth2_client.model.AccessToken
+        """
+        return AccessToken(
+            application=self.app,
+            token=raw_token['access_token'],
+            expires=self.get_expires(raw_token),
+            scope=raw_token.scope,
+            raw_token=dict(raw_token),
+            token_type=raw_token['token_type']
+        )
+
+    def fetch_token_client_credentials(self):
```

Comment:
does it make sense to move it to Oauth2 client?

Reply:
We will ultimately know only when there is another provider to implement using the same flow. But you are right, looks generic enough. Moving.